lib/cadet/assessments/assessments.ex

```diff
@@ -1470,15 +1471,20 @@ defmodule Cadet.Assessments do
         })
         |> Repo.one()
 
-      xp = decimal_to_integer(total.total_xp)
-
       cur_time =
         if submission.submitted_at == nil do
           Timex.now()
         else
           submission.submitted_at
         end
 
+      xp =
+        if is_integer(total.total_xp) do
+          total.total_xp
+        else
+          decimal_to_integer(total.total_xp)
+        end
```

Comment:
If it's not an integer, what will it be?

Reply:
This check was added because the code was failing test cases. In the test cases, total.total_xp is seen as nil which causes an error to occur.

Reply:
But what does it mean, logically? Why would you expect total XP to be nil?

Reply:
Based on the current workflow it will not be nil as when a value is inserted it will be set to 0. However, as xp does not have a not null constraint, it is still possible to be nil so having this redundant check will still be useful. Or would it be better to fix the test cases and set xp to 0?